penny_university_frontend/src/middleware/api.js

```diff
@@ -92,34 +88,34 @@ export default store => next => action => {
   if (typeof endpoint !== 'string') {
     throw new Error('Specify a string endpoint URL.')
   }
-  if (!schema) {
+  if (!responseSchema) {
     throw new Error('Specify one of the exported Schemas.')
   }
   // We will always pass a request, success, and failure action type
   if (!Array.isArray(types) || types.length !== 3) {
     throw new Error('Expected an array of three action types.')
   }
-  if (!types.every(type => typeof type === 'string')) {
+  if (!types.every((type) => typeof type === 'string')) {
     throw new Error('Expected action types to be strings.')
   }
 
-  const actionWith = data => {
-    const finalAction = Object.assign({}, action, data)
+  const actionWith = (data) => {
+    const finalAction = { ...action, ...data }
```

Comment:
TIL this destructuring pattern. Javascript has gone and gotten cool since I was an ember.js monkey.

Reply:
Yeah, I like how JS is evolving. Optional chaining is another cool feature we might want to take advantage of - https://github.com/tc39/proposal-optional-chaining